rasa/core/processor.py

```diff
@@ -172,7 +172,7 @@ async def _update_tracker_session(
                 f"Starting a new session for conversation ID '{tracker.sender_id}'."
             )
 
-            if metadata:
+            if not (SessionStarted in tracker.events) and metadata:
```

Comment:
As @ricwo pointed out to me, the approach we took in #5611 was not really correct.

Here, on the line 176 we manually append `SessionStarted` event. But then the same happens inside `_run_action()` call on line 178. That means that we have two `SessionStarted` injected, one with metadata, one without. 

The correct way would be to pass the metadata further in.

Reply:
I'm having some problems with my internet provider, but the technician is scheduled to come today, as soon he fixes I'll fix that and redo the tests.

Sorry by my first mistake, as I told, it was my first PR, my first time with python, first time with rasa, so I knew something had to go wrong....  :/

My bad...

Reply:
Hello @degiz ,

I was looking into it and the _run_action() does nothing before send to the action server, so RASA never adds the event SessionStart. 

```
async def _run_action(
        self, action, tracker, output_channel, nlg, policy=None, confidence=None
    ) -> bool:
        # events and return values are used to update
        # the tracker state after an action has been taken
        try:
            events = await action.run(output_channel, nlg, tracker, self.domain)
        except ActionExecutionRejection:
            events = [ActionExecutionRejected(action.name(), policy, confidence)]
```

Based on what I understood, it was always the actions server that adds the event of session start to tracker.

I don't know where we could insert deeper, the only other option I see in this case is to create a custom classes from action specific to session start and add the metadata inside it.

Could you check if I'm wrong and give some north to continue?

Thanks!!!

Reply:
Hey @lluchini 

> nothing before send to the action server

You're looking in a right direction. 
`events = await action.run(output_channel, nlg, tracker, self.domain)` for the `ACTION_SESSION_START_NAME` will return empty `SessionStarted`. 

Check `action.py:379`, it create new `SessionStarted `, that's where you can inject the metadata.

Reply:
Also, you can attach debugger and see the flow where the new session starts, you should see the code above being executed.

Reply:
Hi @degiz ,

sorry about the delay on getting back on this, I was having a hard time making the deploy of my project on Docker.

I did what you guys suggested (at least I think I did), but I had to change the base method on Action class to include a new parameter.

I already passed, lint, black, pytest and etc....

Please let me know if I should change anything else.

Thanks.